dags/hivemind_etl_helpers/src/db/mediawiki/mediawiki_reader.py

```diff
@@ -0,0 +1,43 @@
+"""Simple reader that reads mediawiki."""
+
+from typing import Any, List, Optional
+
+import wikipedia
+from llama_index.legacy.readers.base import BasePydanticReader
+from llama_index.legacy.schema import Document
+
+
+class MediaWikiReader(BasePydanticReader):
+    """WikiMedia reader.
+
+    Reads a page.
+
+    """
+
+    is_remote: bool = True
+
+    def __init__(self, api_url: Optional[str] = None) -> None:
+        """Initialize with parameters."""
+        if api_url:
+            wikipedia.set_api_url(api_url)
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "WikipediaReader"
+
+    def load_data(self, pages: List[str], **load_kwargs: Any) -> List[Document]:
+        """Load data from the input directory.
+
+        Args:
+            pages (List[str]): List of pages to read.
+
+        """
+        import wikipedia
+
+        results = []
+        for page in pages:
+            wiki_page = wikipedia.page(page, **load_kwargs)
+            page_content = wiki_page.content
+            page_id = wiki_page.pageid
+            results.append(Document(id_=page_id, text=page_content))
```

Comment:
Does the `page` contain a link to it?
If so, please consider adding it to the document' metadata and exclude it from both LLM and embedding model. This is because we want to be able to track the pages link in case of debugging

```
doc = Document(
    id_=page_id,
    text=page_content,
    metadata={
        "url": page_url,
    },
    excluded_embed_metadata_keys=["url"],
    excluded_llm_metadata_keys=["url"],
)
```